packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -137,7 +145,7 @@ const FocusScope = React.forwardRef((props, forwardedRef) => {
         }, 0);
       };
     }
-  }, [container, onMountAutoFocus, onUnmountAutoFocus, focusScope]);
+  }, [container, unmountAutoFocusElement, onMountAutoFocus, onUnmountAutoFocus, focusScope]);
```

Comment:
Something else has changed here semantically because of the new var which now became a dep.
The side effect of focusing back out should really only be when unmounting, but now it's different because it would execute if someone changed the passed `unmountAutoFocusElement`.